### ISSUE TITLE
private registry .dockercfg

### ISSUE THREAD

**Issue by cobrowserAlex** [runseb/machine#44]
To be able to access our private registry over I need to add a .dockercfg in /var/lib/kubelet/ this however gets wiped on every restart of the VM. I am using VirtualBox...

Is there any way we can make this persist?

**Comment by sebgoa** [comment]
@cobrowserAlex thanks for reporting this. let me ping @cab105 so we can look at it

**Comment by cab105** [comment]
Hmmm my initial post never really made it.  Let me try again.  I apologize for taking so long to notice this.

Hello,

The boot2k8s image is a read-only image without any form of persistent mount point.  The current workaround I can see is to generate a new boot2k8s image to include the .dockercfg file at that location.  For a couple of alternatives:
1. Modify boot2k8s to mount a persistent storage device at boot time which should work provided the .dockercfg doesn't have to be in /var/lib/kubelet
2. Update kmachine to allow for passing in a configuration script that will provide user-specific setup details so when the VM is brought down and started back up again via kmachine, then the configuration will be persisted.

I can see option 2 coming in more handy for other environments and providers besides boot2k8s.

Thoughts?
Chris

**Comment by cobrowserAlex** [comment]
As a workaround I now just manually run a script inside the shared directory that copies stuff in the right place, Similar to option 2.
